test-data/unit/pythoneval.test

```diff
@@ -1458,3 +1458,48 @@ _testNoCrashOnGenericUnionUnpacking.py:10: error: Revealed type is 'Union[builti
 _testNoCrashOnGenericUnionUnpacking.py:11: error: Revealed type is 'Union[builtins.str, builtins.int]'
 _testNoCrashOnGenericUnionUnpacking.py:15: error: Revealed type is 'Union[builtins.int*, builtins.str*]'
 _testNoCrashOnGenericUnionUnpacking.py:16: error: Revealed type is 'Union[builtins.int*, builtins.str*]'
+
+[case testEnumIteration]
+from enum import Enum
+class E(Enum):
+    A = 'a'
+l = [e for e in E]
```

Comment:
Also reveal type of `list(E)`?

Reply:
The test is more concerned with the element type of the comprehension, not the resulting list.  I'll edit the test to make that more explicit.

Reply:
> The test is more concerned with the element type of the comprehension, not the resulting list.

It is (almost) never bad to test more.

Reply:
I was looking at it more in terms of trying to check the enum behavior with minimal reliance on other type-checking features.  But I can add the additional check if you feel strongly that it would be valuable.